Update Chinese and Cantonese localization

auroursa opened the pull request:
This PR includes the latest translations for zh-CN, zh-TW and zh-HK (Cantonese), and cleans up outdated strings.
cc @KuwaLee @cirx1e 

Closes #5801

auroursa commented:
I think we are ready to go.

auroursa commented:
Update translations from upstream:
- Add graphic media self label #5758
- Logo by ... #5788
- Remove double 'an' & Re-translation #5782

auroursa commented:
This will also fixed issue #5801.

surfdude29 commented:
> This will also fixed issue #5801.

GitHub pro tip: if you update the description to say that (or use one of the other keywords), the linked issue will be automatically closed when this PR is merged :)

More info:

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

auroursa commented:
> > This will also fixed issue #5801.
> 
> GitHub pro tip: if you update the description to say that (or use one of the other keywords), the linked issue will be automatically closed when this PR is merged :)
> 
> More info:
> 
> en: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests
> 
> zh: https://docs.github.com/zh/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

Thanks a lot! I have updated the description.😄

surfdude29 commented:
No worries! :)

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -1113,18 +1119,22 @@ msgstr "在此輸入寄送至您電子郵件地址的驗證碼。"
 
+#: src/view/com/composer/labels/LabelsBtn.tsx:110
+msgid "Choose self-labels that are applicable for the media you are posting. If none are selected, this post is suitable for all audiences."
+msgstr "請勾選您認為最適用於此媒體內容的標記。若不勾選，則預設為全年齡向。"
```
```suggestion
msgstr "請勾選您認為最符合此媒體內容的標記。若不勾選，則預設為全年齡向。"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -1928,18 +1969,14 @@ msgstr "持續時間："
-msgstr "例如：張藍天"
+#: src/screens/Profile/Header/EditProfileDialog.tsx:318
+msgid "e.g. Alice Lastname"
+msgstr "例如：愛麗絲"
```
```suggestion
msgstr "例如：張藍天"
```

範例應該可以保留？

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -2297,7 +2328,11 @@ msgstr "展開用戶清單"
-#: src/view/screens/NotificationsSettings.tsx:83
+#: src/lib/api/index.ts:356
+msgid "Expected uri to resolve to a record"
+msgstr "預計 URL 將解析為記錄"
```
```suggestion
msgstr "URI 應解析為記錄"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -3444,14 +3488,22 @@ msgstr "登入或註冊"
 
+#: src/view/shell/desktop/RightNav.tsx:104
+msgid "Logo by <0/>"
+msgstr "此 LOGO 由 <0/> 繪製"
```
```suggestion
msgstr "此 Logo 由 <0/> 繪製"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -3444,14 +3488,22 @@ msgstr "登入或註冊"
+
+#: src/view/shell/Drawer.tsx:293
+msgid "Logo by <0>@sawaratsuki.bsky.social</0>"
+msgstr "此 LOGO 由 <0>@sawaratsuki.bsky.social</0> 繪製"
```
```suggestion
msgstr "此 Logo 由 <0>@sawaratsuki.bsky.social</0> 繪製"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -3489,11 +3541,15 @@ msgstr "管理您靜音的文字和標籤"
 
+#: src/view/com/composer/labels/LabelsBtn.tsx:221
+msgid "Media that may be disturbing or inappropriate for some audiences."
+msgstr "可能會令某些受眾感到不安或不合適的媒體內容。"
```
```suggestion
msgstr "可能會使某些受眾感到不安或造成不適的媒體內容。"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -4195,24 +4252,28 @@ msgid "Open conversation options"
-#: src/view/screens/Settings/index.tsx:702
+#: src/view/com/util/post-embeds/ExternalLinkEmbed.tsx:78
+msgid "Open link to {niceUrl}"
+msgstr "開啟 {niceUrl} 的鏈接"
```
```suggestion
msgstr "開啟 {niceUrl} 的連結"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -4224,60 +4285,64 @@ msgstr "開啟導覽"
 
+#: src/view/com/composer/labels/LabelsBtn.tsx:59
+msgid "Opens a dialog to add a content warning to your post"
+msgstr "開啟對話窗來向您的貼文添加內容警告"
```
```suggestion
msgstr "開啟對話窗來向您的貼文加入內容警告"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -4646,6 +4709,10 @@ msgstr "@{0} 的貼文"
 
+#: src/lib/api/index.ts:121
+msgid "Post failed to upload. Please check your Internet connection and try again."
+msgstr "貼文上傳失敗。請檢查您的網路連線並重試。"
```
```suggestion
msgstr "貼文發佈失敗。請檢查您的網路連線並重試。"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -4685,12 +4752,16 @@ msgstr "貼文已釘選"
-#: src/components/TagMenu/index.tsx:266
+#: src/lib/api/index.ts:106
+msgid "Posting..."
+msgstr "貼文發佈中..."
```
```suggestion
msgstr "正在發佈貼文..."
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -5624,11 +5681,11 @@ msgstr "發送一個妙趣的網站！"
 #: src/components/dialogs/VerifyEmailDialog.tsx:189
 msgid "Send Confirmation"
-msgstr ""
+msgstr "發送確認"
```
```suggestion
msgstr "發送確認信"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -6283,7 +6336,7 @@ msgstr "文字輸入框"
 #: src/components/dialogs/VerifyEmailDialog.tsx:82
 msgid "Thank you! Your email has been successfully verified."
-msgstr ""
+msgstr "謝謝你！您的電子郵件地址已成功驗證。"
```
```suggestion
msgstr "謝謝您！您的電子郵件地址已成功驗證。"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -6452,18 +6499,28 @@ msgstr "取得列表時發生問題，點擊這裡重試。"
+#: src/view/com/posts/FeedShutdownMsg.tsx:71
+#: src/view/screens/ProfileFeed.tsx:211
+msgid "There was an issue updating your feeds, please check your internet connection and try again."
+msgstr "更新動態源時出現問題，請檢查您的網路連線並重試。"
```
```suggestion
msgstr "更新動態時出現問題，請檢查您的網路連線並重試。"
```

cirx1e commented on src/locale/locales/zh-CN/messages.po:
```diff
@@ -2406,6 +2441,11 @@ msgstr "无法保存这张图片：{0}"
+#: src/lib/api/index.ts:145
+#: src/lib/api/index.ts:170
+msgid "Failed to save post interaction settings. Your post was created but users may be able to interact with it."
+msgstr "无法保存帖文互动选项。您的帖文已成功创建，但其他用户可能无法与其互动。"
```
```suggestion
msgstr "无法保存帖文互动选项。您的帖文已成功创建，但其他用户可能仍然能够与其互动。"
```

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -2406,6 +2441,11 @@ msgstr "無法儲存圖片：{0}"
+#: src/lib/api/index.ts:145
+#: src/lib/api/index.ts:170
+msgid "Failed to save post interaction settings. Your post was created but users may be able to interact with it."
+msgstr "無法儲存貼文互動設定。您的貼文已創建，但其他用戶可能無法與其互動。"
```
```suggestion
msgstr "無法儲存貼文互動限制。您的貼文已發佈，但其他用戶可能仍然能夠與其互動。"
```

auroursa commented:
We are ready to go (again).

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -2625,7 +2625,7 @@ msgstr "已被您跟隨的 <0>{0}</0> 和{1, plural, one {其他 # 人跟隨} ot
 #: src/components/KnownFollowers.tsx:204
 msgid "Followed by <0>{0}</0> and <1>{1}</1>"
-msgstr "已被您跟隨的 <0>{0}</0> 和 <1>{1}</1> 跟隨"
+msgstr "已被您跟隨的 <0>{0}</0> 和<1>{1}</1> 跟隨"
```
```suggestion
msgstr "已被您跟隨的 <0>{0}</0> 和 <1>{1}</1> 跟隨"
```

https://github.com/bluesky-social/social-app/blob/fd2e94f3534ea459b08a4d6b9a6f0aaccfd2445e/src/components/KnownFollowers.tsx#L203-L213
`{0}` 和 `{1}` 看起來都是使用者的顯示名稱，前後應加上空格

auroursa replied:
这边<1>{1}<1>前面不加空格是出于美观考虑，因为这里实际显示的内容为“其他*人”，如果前面有空格就会变成“及 其他*人 跟随”

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -4060,6 +4064,10 @@ msgstr "未找到符合 {query} 的結果"
 
+#: src/view/com/composer/labels/LabelsBtn.tsx:129
+msgid "No self-labels can be applied to this post because it contains no media."
+msgstr "這篇貼文沒有任何標記，因為不包含媒體內容。"
```
```suggestion
msgstr "這則貼文不含任何媒體內容，因此無法加入標記。"
```

https://github.com/bluesky-social/social-app/blob/fd2e94f3534ea459b08a4d6b9a6f0aaccfd2445e/src/view/com/composer/labels/LabelsBtn.tsx#L116-L135

按下「新增內容警告」後顯示的提示訊息

cirx1e commented on src/locale/locales/zh-CN/messages.po:
```diff
@@ -4060,6 +4064,10 @@ msgstr "未找到 {query} 的结果"
 
+#: src/view/com/composer/labels/LabelsBtn.tsx:129
+msgid "No self-labels can be applied to this post because it contains no media."
+msgstr "这个帖文没有任何标记，因为帖文中不包含媒体内容。"
```
```suggestion
msgstr "这个帖文不包含媒体内容，因此无法添加标记。"
```

https://github.com/bluesky-social/social-app/blob/fd2e94f3534ea459b08a4d6b9a6f0aaccfd2445e/src/view/com/composer/labels/LabelsBtn.tsx#L116-L135

点击「新增内容警告」后显示的提示讯息

flower-elf commented on src/locale/locales/zh-CN/messages.po:
```diff
@@ -3125,29 +3167,29 @@ msgstr "邀请码：1 个可用"
 #: src/screens/StarterPack/Wizard/StepDetails.tsx:32
 msgid "Invites, but personal"
-msgstr "邀请，但保持私密"
+msgstr "个性化邀请朋友加入"
```
可能“以个性化的方式邀请朋友加入！”会更好

flower-elf reviewed (commented):
Some Suggestions

auroursa commented:
> Some Suggestions

Thank you!

notlin4 commented:
不太建議將顯示名稱修改為暱稱，暱稱的意思是親暱的稱呼，而顯示名稱是使用者選擇在平台上使用的名稱，兩者概念不同

flower-elf commented:
> 不太建議將顯示名稱修改為暱稱，暱稱的意思是親暱的稱呼，而顯示名稱是使用者選擇在平台上使用的名稱，兩者概念不同

在简体中文语境中这个词，相较于显示名称会更好理解一些
因为中国大陆境内有不少的软件都是用了这个词，所以希望能够做到一看就懂
如果昵称确实会导致理解问题，那么直接更新为名称也是可以的

auroursa commented:
> 不太建議將顯示名稱修改為暱稱，暱稱的意思是親暱的稱呼，而顯示名稱是使用者選擇在平台上使用的名稱，兩者概念不同

我也在顾虑这点，因为台湾繁中语境下“暱稱”没有类似的意思，所以如果要改变 Display Name 的翻译，就会保持繁中不变只改简中

但这种名词上存在差异，可能会影响后续沟通；例如一位简中用户说“昵称”，但繁中还是“顯示名稱”，导致两边都无法相互理解意思。

flower-elf commented:
> > 不太建議將顯示名稱修改為暱稱，暱稱的意思是親暱的稱呼，而顯示名稱是使用者選擇在平台上使用的名稱，兩者概念不同
> 
> 我也在顾虑这点，因为台湾繁中语境下“暱稱”没有类似的意思，所以如果要改变 Display Name 的翻译，就会保持繁中不变只改简中
> 
> 但这种名词上存在差异，可能会影响后续沟通；例如一位简中用户说“昵称”，但繁中还是“顯示名稱”，导致两边都无法相互理解意思。

如果出于该种考虑，可能直接“名称”会好些

auroursa commented:
梳理了一下，目前简中和繁中用“名称”代指的只有 App 专用密码、列表名称，把 Display Name 直接翻译为名称应该不会产生混淆

那我先把简中的 Display Name 翻译改为“名称”，繁中我想再问问繁中社群的看法再修改

auroursa commented:
梳理了一下，目前简中和繁中用“名称”代指的只有 App 专用密码命名，和列表名称。把 Display Name 直接翻译为名称应该不会产生混淆，应该没有问题

auroursa commented:
Ready for merge.

cirx1e commented:
fixed a translation error
- https://github.com/KuwaLee/bluesky-social-app/pull/10

cirx1e commented on src/locale/locales/zh-TW/messages.po:
```diff
@@ -3167,15 +3167,15 @@ msgstr "邀請碼：1 個可用"
 #: src/screens/StarterPack/Wizard/StepDetails.tsx:32
 msgid "Invites, but personal"
-msgstr "邀請，但僅限個人"
+msgstr "邀請朋友加入，同時保持個人化"
```
```suggestion
msgstr "邀請，但更具個人風格"
```

auroursa commented:
@cirx1e Thank you!

notlin4 commented:
台灣較常使用對話框或對話方塊，而非對話窗

auroursa commented:
> 台灣較常使用對話框或對話方塊，而非對話窗

感谢，先前的翻译是「對話框」與「對話窗」混用，由于「對話窗」的 review 时间较晚，就全部改成了后者，稍后我提交个 commit 会全部改回「對話框」

auroursa commented:
@notlin4  I have fixed it in 2b123d2

notlin4 commented:
我剛剛開了個 [PR](https://github.com/KuwaLee/bluesky-social-app/pull/11)，修正了一些一致性問題，改進了一些翻譯等等

auroursa commented:
@cirx1e @KuwaLee Could you please review this [PR](https://github.com/KuwaLee/bluesky-social-app/pull/11)?

It mainly involves translation improvements for zh-TW, and I would like to confirm it.

pfrazee reviewed (approved):
Thank you @auroursa !!
